model/generate.go

```diff
@@ -275,17 +276,57 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, v *Version
 	return nil
 }
 
-func (g *GeneratedProject) findVariantsToActivate() []string {
-	var toActivate []string
+type batchTimeTasksAndVariants struct {
+	tasks    map[string][]string // tasks by variant that have batchtime
+	variants []string            // variants that have batchtime
+}
+
+func newBatchTimeTasksAndVariants() batchTimeTasksAndVariants {
+	return batchTimeTasksAndVariants{
+		tasks:    map[string][]string{},
+		variants: []string{},
+	}
+}
+
+func (b *batchTimeTasksAndVariants) isBatchTimeVariant(variant string) bool {
+	return utility.StringSliceContains(b.variants, variant)
+}
+
+func (b *batchTimeTasksAndVariants) batchTimeTasks(variant string) []string {
+	return b.tasks[variant]
+}
+
+func (b *batchTimeTasksAndVariants) hasAnyBatchTimeTasks() bool {
+	for _, tasks := range b.tasks {
```

Comment:
Would just doing len(b.tasks) be sufficient? Or can there be variants in the map whose value is an empty array?

Reply:
I can change findTasksAndVariantsWithBatchTime so that the empty array isn't possible